kong/core/router.lua

```diff
@@ -619,30 +621,18 @@ function _M.new(apis)
     end
 
 
-    upstream_host = api_t.upstream_host
-
-
     if api_t.strip_uri_regex then
       local stripped_uri = re_sub(uri, api_t.strip_uri_regex, "/$1", "jo")
       ngx.req.set_uri(stripped_uri)
     end
 
 
-    if api_t.preserve_host then
-      if not headers then
-        headers = ngx.req.get_headers()
-      end
-
-      upstream_host = headers["host"]
-    end
```

Comment:
There is a test in the unit tests that currently ensures this behavior is taken care of by `exec`. It it now failing.